dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/TestUtils/TestAssistantsOpenAIClient.cs

```diff
@@ -34,105 +34,104 @@ public void Reset()
+                    Guid.NewGuid().ToString(),
+                    DateTimeOffset.Now,
+                    newThread.Id,
+                    MessageRole.User,
+                    new List<MessageContent>() { AssistantsModelFactory.MessageTextContent(m.Content, annotations) },
+                    Assistant.Id,
+                    null,
+                    m.FileIds,
+                    null
+                );
+                messages.Add(newMessage);
+            }
+
+            Messages.Add(newThread.Id, messages);
+            Runs.Add(newThread.Id, new List<ThreadRun>());
+            return Task.FromResult(Response.FromValue(newThread, Mock.Of<Response>()));
         }
 
-        public override Task<Message> CreateMessageAsync(string threadId, MessageCreateParams messageCreateParams, CancellationToken cancellationToken)
+        public override Task<Response<ThreadMessage>> CreateMessageAsync(string threadId, MessageRole role, string content, IEnumerable<string> fileIds = null, IDictionary<string, string> metadata = null, CancellationToken cancellationToken = default)
         {
-            Message newMessage = new()
-            {
-                Id = Guid.NewGuid().ToString(),
-                CreatedAt = DateTime.Now.Ticks / 1000,
-                Content = new() { new() { Type = "text", Text = new() { Value = messageCreateParams.Content } } },
-                Role = messageCreateParams.Role,
-                Metadata = messageCreateParams.Metadata,
-                FileIds = messageCreateParams.FileIds ?? new()
-            };
+            var annotations = new List<MessageTextAnnotation>() { AssistantsModelFactory.MessageTextAnnotation() };
+            ThreadMessage newMessage = AssistantsModelFactory.ThreadMessage(
+                Guid.NewGuid().ToString(),
+                DateTimeOffset.Now,
+                threadId,
+                role,
+                new List<MessageContent> { AssistantsModelFactory.MessageTextContent(content, annotations) },
+                Assistant.Id,
+                null,
+                fileIds,
+                null
+            );
+
             if (Messages.ContainsKey(threadId))
             {
                 Messages[threadId].Add(newMessage);
             }
             else
             {
                 Messages.Add(threadId, new() { newMessage });
-            }
+            };
 
-            return Task.FromResult(newMessage);
+            return Task.FromResult(Response.FromValue(newMessage, Mock.Of<Response>()));
         }
 
-        public override async IAsyncEnumerable<Message> ListNewMessagesAsync(string threadId, string? lastMessageId, [EnumeratorCancellation] CancellationToken cancellationToken)
+        public override async Task<Response<PageableList<ThreadMessage>>> GetMessagesAsync(string threadId, int? limit = null, ListSortOrder? order = null, string after = null, string before = null, CancellationToken cancellationToken = default)
```

Comment:
Could you handle these warnings listed below?

Reply:
fixed